src/requirements/Payment/PaymentForm.tsx

```diff
@@ -62,9 +62,13 @@ const PaymentForm = ({
       setValue(`${baseFieldPath}.data.id`, registeredVaultId),
   })
 
+  const isRequirementAdded = useRef(false)
   useEffect(() => {
     if (!vaultId) return
-    addRequirement()
+    if (!isRequirementAdded.current) {
```

Comment:
I wonder if we could completely get rid of this `useEffect`? Maybe if we set `${baseFieldPath}.data.id` in `useRegisterVault`'s `onSuccess`, and call `addRequirement` there too? Not sure if that'd work properly, but I think we could try something like that, it'd simplify the code a lot. 😉